src/Chronos.php

```diff
@@ -563,27 +563,21 @@ public static function create(
         ?int $year = null,
         ?int $month = null,
         ?int $day = null,
-        ?int $hour = null,
-        ?int $minute = null,
-        ?int $second = null,
-        ?int $microsecond = null,
+        ?int $hour = 0,
+        ?int $minute = 0,
+        ?int $second = 0,
+        ?int $microsecond = 0,
         DateTimeZone|string|null $timezone = null
     ): static {
         $now = static::now();
         $year = $year ?? (int)$now->format('Y');
         $month = $month ?? $now->format('m');
         $day = $day ?? $now->format('d');
 
-        if ($hour === null) {
-            $hour = $now->format('H');
-            $minute = $minute ?? $now->format('i');
-            $second = $second ?? $now->format('s');
-            $microsecond = $microsecond ?? $now->format('u');
-        } else {
-            $minute = $minute ?? 0;
-            $second = $second ?? 0;
-            $microsecond = $microsecond ?? 0;
-        }
+        $hour = $hour ?? $now->format('H');
+        $minute = $minute ?? $now->format('i');
+        $second = $second ?? $now->format('s');
+        $microsecond = $microsecond ?? $now->format('u');
```

Comment:
Having all the time components default to `0` will be a bigger break than defaulting all values to 'now'. Would we be in a better place if we removed the special scenario for `hour: 0`? Having each component of the time default to the present, would make using named parameters preserve as much compatibility with what we have today, and increase consistency between how the date & time component parameters work.

Reply:
The reason for this change is to allow calls like `Chronos::create(2023, 01, 01, minute: 10)` Currently that will create an instance where hour, second and microsecond are not and minute is 10.